src/searching/binary_search.rs

```diff
@@ -0,0 +1,220 @@
+use std::cmp::Ordering;
+/// ## Binary Search algorithm.
+/// Iterating over the array by dividing half of the array and returns the index value of the target element in the array.
+/// you can pass both ascending and decending array respectively.
```

Comment:
I think we might not want this feature since it is not core to the algorithm, so we might not want to include it in documentation, nor have it in the implementation.

You can imagine that if you have a binary search implementation that only works for non-decreasing order collections and you want to find an element in an array sorted in non-increasing order, you would just have to reverse the array before passing it to the algorithm.

Reply:
nit: Also, new sentences should start with a capital letter.